jetty-server/src/main/java/org/eclipse/jetty/server/HttpConnection.java

```diff
@@ -302,15 +300,19 @@ else if (filled < 0)
         {
             if (LOG.isDebugEnabled())
                 LOG.debug("{} caught exception {}", this, _channel.getState(), x);
-            BufferUtil.clear(_requestBuffer);
-            releaseRequestBuffer();
+            // Force the release of the _retainableByteBuffer.
+            if (_retainableByteBuffer != null)
+            {
+                while (!_retainableByteBuffer.release());
+                _retainableByteBuffer = null;
+            }
```

Comment:
Please let's go to `clear()` + `releaseRequestBuffer()` like it was before and it is in other places such as `WebSocketConnection`.
Please make sure `EndPoint.close(x)` is called even if `releaseRequestBuffer()` throws.

Reply:
Ok.